app/forms/supplementary/GoodsLocation.scala

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package forms.supplementary
+
+import play.api.data.Form
+import play.api.data.Forms
+import play.api.data.Forms.{optional, text}
+import play.api.libs.json.Json
+import services.Countries.allCountries
+import utils.validators.FormFieldValidator._
+
+case class GoodsLocation(
+  country: Option[String],
+  typeOfLocation: Option[String],
+  qualifierOfIdentification: Option[String],
+  identificationOfLocation: Option[String],
+  additionalIdentifier: Option[String],
+  streetAndNumber: Option[String],
+  postCode: Option[String],
+  city: Option[String]
+)
+
+object GoodsLocation {
+  implicit val format = Json.format[GoodsLocation]
+
+  val formId = "GoodsLocation"
+
+  val mapping = Forms.mapping(
+    "country" -> optional(
+      text()
+        .verifying("supplementary.country.empty", _.trim.nonEmpty)
+        .verifying(
+          "supplementary.country.error",
+          input => input.isEmpty || allCountries.filter(country => country.countryName == input).nonEmpty
```

Comment:
I am wondering if it would be better to use `input.trim.isEmpty` ?

Reply:
Yup